### PR TITLE
Add some convenience funcs for Tx tracing

### PR DESCRIPTION
**Pull request by bmarini**

### CODE REVIEW

File: tx.go [hunk 1]
```diff
@@ -123,6 +127,51 @@ func FromContext(ctx context.Context) (Tx, bool) {
...
+
+	return ctx, func() {
+		tx.End()
+	}
+}
+
+// TraceExternal adds an external segment to the newrelic transaction, if one exists in the context.
+func TraceExternal(ctx context.Context, host, name string) DoneFunc {
+	return trace(ctx, name, func(tx Tx) {
+		tx.StartExternal(host, name)
```
Comment (sanjayprabhu) [inline tx.go]: What if these return errors? We may want the DoneFunc to propagate any errors to us. We shouldn't fail, but the calling code may (and probably should) choose to log it.

Comment (bmarini) [inline tx.go]: Good catch

File: tx.go [hunk 2]
```diff
@@ -123,6 +127,52 @@ func FromContext(ctx context.Context) (Tx, bool) {
...
+//     defer done()
+func TraceRequest(ctx context.Context, name string, req *http.Request) (context.Context, DoneFunc, error) {
+	tx := NewRequestTx(name, req.URL.String())
+	ctx = WithTx(ctx, tx)
+	err := tx.Start()
+	df := func() error {
+		return tx.End()
+	}
+
+	return ctx, df, err
```
Comment (sanjayprabhu) [inline tx.go]: I was actually thinking these top level functions need only return `DoneFunc`s and we just wouldn't call the "end" functions if start failed. That way the callers have a simpler api. They don't have to check the error twice (not like they can do much if start failed anyway, they're not going to abort their actual processing).

Something like:
```go
func TraceRequest(...) (.., DoneFunc) {
   err := tx.Start()
   df := func() error {
      if err {
         return err
      } else {
         return tx.End()
      }
   }
}
```

Comment (bmarini) [inline tx.go]: That makes sense. Much simpler

Comment (bmarini) [inline tx.go]: Actually, it'd probably be better to return an interface that has `Done()` and `Err()` methods.

That way:

``` go
t := TraceRequest(...)
defer function() {
    t.Done()
    if t.Err() != nil {
        // log error
    }
}()
```

Comment (sanjayprabhu) [inline tx.go]: Nice :+1:

**Comment by sanjayprabhu** [comment]
Nice :+1:

**Pull request merged**